Applications/FileIO/GocadIO/GocadSGridReader.cpp

```diff
@@ -0,0 +1,855 @@
+{
+    for (auto prop_it(_property_meta_data_vecs.begin());
+         prop_it != _property_meta_data_vecs.end();
+         prop_it++)
+    {
+        std::string const& fname(prop_it->_property_data_fname);
+        if (prop_it->_property_data_fname.empty())
+        {
+            WARN("Empty filename for property %s.",
+                 prop_it->_property_name.c_str());
+            continue;
+        }
+        std::vector<float> float_properties =
+            BaseLib::readBinaryArray<float>(fname, _index_calculator._n_cells);
+        DBUG(
+            "GocadSGridReader::readElementPropertiesBinary(): Read %d float "
+            "properties from binary file.",
+            _index_calculator._n_cells);
+
+        std::transform(float_properties.cbegin(), float_properties.cend(),
+                       float_properties.begin(), [](float const& val) {
+                           return BaseLib::swapEndianness(val);
+                       });
+
+        prop_it->_property_data.resize(float_properties.size());
+        std::copy(float_properties.begin(), float_properties.end(),
+                  prop_it->_property_data.begin());
+        if (prop_it->_property_data.empty())
+        {
+            ERR("Reading of element properties file '%s' failed.",
+                fname.c_str());
+        }
+    }
+}
+
+std::vector<int> GocadSGridReader::readFlagsBinary() const
+{
+    std::vector<int> result;
+    if (!_double_precision_binary)
+    {
+        result = BaseLib::readBinaryArray<int32_t>(_flags_fname,
+                                                   _index_calculator._n_nodes);
+        std::for_each(result.begin(), result.end(),
+                      [](int32_t& val) { BaseLib::swapEndianness(val); });
+    }
+    else
+    {
+        result = BaseLib::readBinaryArray<int>(_flags_fname,
+                                               _index_calculator._n_nodes);
+        std::for_each(result.begin(), result.end(),
+                      [](int& val) { BaseLib::swapEndianness(val); });
+    }
+
+    if (result.empty())
+        ERR("Reading of flags file '%s' failed.", _flags_fname.c_str());
+
+    return result;
+}
+
+std::vector<Bitset> GocadSGridReader::readRegionFlagsBinary() const
+{
+    std::vector<Bitset> result;
+
+    std::ifstream in(_region_flags_fname.c_str());
+    if (!in)
+    {
+        ERR("readRegionFlagsBinary(): Could not open file '%s' for input.\n",
+            _region_flags_fname.c_str());
+        in.close();
+        return result;
+    }
+
+    std::size_t const n = _index_calculator._n_nodes;
+    result.resize(n);
+
+    std::size_t k = 0;
+    while (in && k < n)
+    {
+        result[k++] = readBits(in, regions.size());
+    }
+    if (k != n && !in.eof())
+        ERR("Read different number of values. Expected %d, got %d.\n", n, k);
+
+    return result;
+}
+
+void GocadSGridReader::createElements(
+    std::vector<MeshLib::Node*> const& nodes,
+    std::vector<MeshLib::Element*>& elements) const
```

Comment:
The function has no other effects, but to change the `elements` parameter. Better to return, than to use an output-parameter.